opencog/query/PatternMatchCallback.h

```diff
@@ -238,6 +238,16 @@ class PatternMatchCallback
 		                       const std::map<Handle, Handle> &term_soln) = 0;
 
 		/**
+		 * Called when we are comparing unordered links. It would be useful
+		 * if we want to skip this grounding and move on to the next right away
+		 * without having to compare all of its possible permutations.
+		 */
```

Comment:
I was only looking at what your code actually did, not what you want it to do.  By looking at what it actually did, I suggested a different way of writing it.   What your code did was to completely skip and always fail the grounding of unordered links.